### ISSUE TITLE
PostScriptCompiler compiled result violates the Content security policy (no unsafe-eval) 

### ISSUE THREAD

**Issue by longsleep** [mozilla/pdf.js#5573]
https://github.com/mozilla/pdf.js/blob/45ca95370513c3c5edfefa802e6e6152bec56a84/src/core/function.js#L406

is using  return new Function('src', 'srcOffset', 'dest', 'destOffset', compiled);

This does not work when using a CSP for security reasons which does not allow unsafe-eval for script-src.

PDFJS.version = '1.0.907';

**Comment by TheKidCoder** [comment]
I would like to raise this issue again as we just ran into it inside of a web worker. We hit this compile issue because we do not allow `unsafe-eval`. Oddly enough, this issue only cropped up when Chrome 59 landed. They might have changed enforcement with CSP & Web Workers. We were able to fix the issue by simply adding this line:
```javascript
if (!!PDFJS.isEvalSupported && compiled) {
  return new Function('src', 'srcOffset', 'dest', 'destOffset', compiled);
}
```
to https://github.com/mozilla/pdf.js/blob/7b4887dd211c44d5ef852ad019921fb41fa3709b/src/core/function.js#L403

This seems to work for us. I will create a pull request.
